### DISCUSSION TITLE
Using existing db and keys

### DISCUSSION THREAD

**Discussion by ozanbaba** [dani-garcia/vaultwarden#1907] [Q&A] [answered]
Hello All,

I had a bitwarde_rs system up and running. I had to keep it old while waiting for debian 11 release. After the realese, I updated the server and everything in it and of course bitwarden_rs. Now I have vaultwarden up and running. However I messed up the update and lost the database files. As I had full system backup before the update, I have recovered the database files successfully. 

My questions is how I can use those database and key files in vaultwarden system.
Information about my setup: I pull source code from vaultwarden repo and build it myself and then run it. I didnt use docker at all.

Regards,
Ozan

**Comment by jjlin** [comment]
See https://github.com/dani-garcia/vaultwarden/wiki/Backing-up-your-vault#restoring-backup-data.

**Comment by ozanbaba** [comment, marked as answer]
Thank you for reply, jjlin.
I'm writing down what I exectly did for future people.

- I uploaded the data folder as tarball to the server.
- I extracted the tarball in place I want my data folder is
- I created any missing folders inside the data folder (sends and attachements)
- I created .env file using the template
- I pointed each data folder entry to the data folder to my files
- I pointed key file to key_file.pem file

Started vaultwarden and it worked.
